### ISSUE TITLE
Option help text and error messages do not clearly explain the (indexed) relationship between --format and --out

### ISSUE THREAD

**Issue by dkarlovi** [Behat/Behat#1409]
```
behat --format=pretty --format=junit --out=some/dir
```

This yields an exception because the `--out` will actually apply to the first formatter `pretty`, not the `junit` one, creating UX issues and confusion.

An alternative is to merge the params like Docker does:

```
behat --format=pretty --format=junit,target=some/dir
```

**Comment by swaibar** [comment]
you can simply use std as the output:
```
behat --format=pretty --out=std --format=junit --out=some/dir
```

**Comment by carlos-granados** [comment]
Agree that seeing your example the behaviour seems weird but changing it would be a breaking change. And trying to tie both options together might be difficult and I am sure there will be weird cases. @acoulton I am tempted to close this one given that there is an easy option to get the needed result. What do you think?

**Comment by acoulton** [comment]
@carlos-granados I agree, I'd rather not take responsibility for parsing values out of composite arguments and the edge cases that might involve. The behaviour does seem strange if you're not expecting it, but I've used other CLI tools that have a similar approach. I also just checked and [this aspect of the CLI is actually documented](https://github.com/Behat/docs/blob/45806f68d7d8b7c496ee49fcb14be7d31c40b363/user_guide/command_line_tool/formatting.rst?plain=1#L48-L56) so I think it's OK.

**Comment by stof** [comment]
Maybe the description of the option should be improved so that `behat --help` explains this relation between both options

**Comment by dkarlovi** [comment]
IMO at least fixing the error message you get here would be beneficial, IIRC I needed to debug what's going on and only retroactively figured out why this is happening. Currently the error message is something completely unrelated.

**Comment by dkarlovi** [comment]
```
$ vendor/bin/behat  --format=pretty --format=junit --out=var/log/

In ConsoleOutputFactory.php line 68:
                                                                         
  Filename expected as `output_path` parameter, but got `/app/var/log`. 
```

**Comment by dkarlovi** [comment]
TBH there's no way a user would be able to figure out what's going on here.

Note, this also doesn't work with the same error:

```
vendor/bin/behat  --format=junit --out var/log/ --format pretty
```

**Comment by acoulton** [comment]
OK, I'll reopen this as a bug limited to:
* Improving the documentation of the --format and --out options to make the relationship clear
* Improving the error message (which may relate to #1408)
